pyro/distributions/torch_distribution.py

```diff
@@ -197,7 +199,10 @@ class TorchDistribution(torch.distributions.Distribution, TorchDistributionMixin
     method to improve gradient estimates and set
     ``.has_enumerate_support = True``.
     """
-    pass
+    # Provides a default `.expand` method for Pyro distributions which overrides
+    # torch.distributions.Distribution.expand (throws a NotImplementedError)
+    def expand(self, batch_shape, _instance=None):
```

Comment:
Do I understand correctly that this must live in `TorchDistribution` rather than `TorchDistributionMixin` due to mro?

Reply:
That's right; otherwise we'll call the default `torch.distributions.Distribution.expand` which will throw a `NotImplementedError`. Maybe we can move this upstream to PyTorch, if it looks reasonable.

Reply:
I have moved this to `TorchDistributionMixin` so that it is also available to `.expand_by` and set `expand = TorchDistributionMixin.expand` here.